drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -82,18 +108,14 @@ local function do_configure(driver, device)
   -- where devices with multiple endpoints with the same device type cannot be detected
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local num_switch_eps = #switch_eps
-  table.sort(switch_eps)
-  --Default MCD switch handling depends on consecutive endpoint numbering
-  if num_switch_eps == switch_eps[num_switch_eps] then
-    if num_switch_eps > 1 then
-      device:try_update_metadata({profile = string.format("switch-%d", math.min(num_switch_eps, MAX_MULTI_SWITCH_EPS))})
-    end
-    if num_switch_eps > MAX_MULTI_SWITCH_EPS then
-      error(string.format(
-        "Matter multi switch device will not function. Profile doesn't exist with %d components",
-        num_switch_eps
-      ))
-    end
+  if num_switch_eps > 1 then
+    device:try_update_metadata({profile = string.format("switch-%d", math.min(num_switch_eps, MAX_MULTI_SWITCH_EPS))})
+  end
+  if num_switch_eps > MAX_MULTI_SWITCH_EPS then
+    error(string.format(
+      "Matter multi switch device will have limited function. Profile doesn't exist with %d components",
+      num_switch_eps
```

Comment:
is it worth setting it to the highest available value?

Reply:
If a device has more than the max number of supported switches, it will switch to the highest available profile on line 112 above. This would mean that only those endpoints that could be mapped would actually be functional, and the excess endpoints would be mapped to the "main" component